### PR TITLE
workflow: sign release images

### PR DESCRIPTION
**Pull request by josedonizetti**
### 1. Explain what the PR does

Fix https://github.com/aquasecurity/tracee/issues/2804
Fix https://github.com/aquasecurity/tracee/issues/2907

Sign release images. 

### 2. Explain how to test it

This was implemented on the [`release-snapshot`](https://github.com/aquasecurity/tracee/blob/main/.github/workflows/release-snapshot.yaml) workflow to validate it works.

You can check the signed dev image with:

```
cosign verify aquasec/tracee:aarch64-dev \
--certificate-identity=https://github.com/aquasecurity/tracee/.github/workflows/release-snapshot.yaml@refs/heads/main \
--certificate-oidc-issuer=https://token.actions.githubusercontent.com | jq
```

### CODE REVIEW

**Review by yanivagman — APPROVED** [review]
LGTM

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -86,3 +108,7 @@ jobs:
           docker image tag tracee:latest aquasec/tracee:${ARCH}-${TAG}
           docker image push aquasec/tracee:${ARCH}
           docker image push aquasec/tracee:${ARCH}-${TAG}
+      - name: Sign Docker image
+        run: |
+          cosign sign -y $(docker inspect --format='{{index .RepoDigests 0}}' aquasec/tracee:${ARCH})
```
Comment (developer-guy) [inline .github/workflows/release.yaml:113]: You should use the "$IMAGE@$DIGEST" format to sign your container image, so the correct form is the following:

```
cosign sign -y aquasec/tracee:${ARCH}@$(docker inspect --format='{{index .RepoDigests 0}}' aquasec/tracee:${ARCH})
```

Comment (developer-guy) [inline .github/workflows/release.yaml:113]: in addition to that, once you push the image, you need to re-pull it to be able to fill the digests field of the configuration of the image, instead, you can use the `crane` tool to find the digests without needing to re-pull the image.

```
cosign sign -y aquasec/tracee:${ARCH}@$(crane digest aquasec/tracee:${ARCH})
```

Comment (developer-guy) [reply .github/workflows/release.yaml:113]: oh, that's great, didn't notice this! Then, it looks correct, yes!

Comment (developer-guy) [reply .github/workflows/release.yaml:113]: this is also great! thanks, didn't know this! Otherwise, everything looks correct to me!

**Review by developer-guy — APPROVED** [review]
not a maintainer but LGTM

**Pull request merged**